wgpu-types/src/lib.rs

```diff
@@ -1502,7 +1759,40 @@ pub struct TextureViewDescriptor<L> {
     pub array_layer_count: Option<NonZeroU32>,
 }
 
+impl <'a> TextureViewDescriptor<Option<Cow<'a, str>>> {
+    pub fn new(format: TextureFormat, dimension: TextureViewDimension) -> Self {
+        Self {
+            label: None,
+            format,
+            dimension,
+            aspect: TextureAspect::default(),
+            base_mip_level: 0,
+            level_count: None,
+            base_array_layer: 0,
+            array_layer_count: None,
+        }
+    }
+
+    pub fn label(&mut self, label: impl IntoCow<'a, str>) -> &mut Self {
+        self.label = Some(label.into_cow());
+        self
+    }
+}
+
 impl<L> TextureViewDescriptor<L> {
+
+    pub fn mip_levels(&mut self, base_mip_level: u32, level_count: Option<NonZeroU32>) -> &mut Self {
+        self.base_mip_level = base_mip_level;
+        self.level_count = level_count;
+        self
+    }
+
+    pub fn array_layers(&mut self, base_array_layer: u32, array_layer_count: Option<NonZeroU32>) -> &mut Self {
+        self.base_array_layer = base_array_layer;
+        self.array_layer_count = array_layer_count;
+        self
+    }
+
```

Comment:
@kvark Could you double check this? I accidentally missed its implementation previously.